### PR TITLE
Setup Jekyll for site generation

### LINKED ISSUE #44: Add a validation rule that verifies Table of Contents
Some SIPs add a Table of Contents, which may become desynced from the paragraphs. We should add a rule that verifies if they still match.

The algorithm for converting headers into `[Link Slugs](#link-slugs)` is not simple and may require some time to figure out all the edges cases.

### PR DESCRIPTION
**Pull request by FrederikBolding**
Sets up Jekyll for site generation. Based on the set up used for EIPs and CAIPs (pending review).

Removes existing tables of content as those can be automatically generated.

fixes #44

### CODE REVIEW

File: _includes/siptable.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+  </style>
+  {% for status in site.data.statuses %}
+    {% assign sips = include.sips|where:"status",status|sort:"sip" %}
+    {% assign count = sips|size %}
+    {% if count > 0 %}
+      <h2 id="{{status|slugify}}">{{status}}</h2>
+      <table class="siptable">
+        <thead>
+          {% if status == "Last Call" %}
```
Comment (ritave) [inline _includes/siptable.html:17]: We don't have a `Last Call` status

File: _layouts/sip.html [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      <th>Created</th>
+      <td>{{ page.created | xml_escape }}</td>
+    </tr>
+    {% if page.updated != undefined %}
+    <tr>
+      <th>Updated</th>
+      <td>{{ page.updated | xml_escape }}</td>
+    </tr>
+    {% endif %}
+    {% if page.requires != undefined %}
```
Comment (ritave) [inline _layouts/sip.html:54]: Multiple properties in preamble that don't exist in SIPs

File: _layouts/sip.html [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+  <div class="toc">
+    <h2>Table of Contents</h2>
+    {% include toc.html html=content sanitize=true h_max=3 %}
+  </div>
+
+  {% include anchor_headings.html html=content anchorClass="anchor-link" beforeHeading=true %}
+
+  <h2>Citation</h2>
+  <p>Please cite this document as:</p>
```
Comment (ritave) [inline _layouts/sip.html:76]: Hah, quite a big ego for us to be thinking someone will be citing Snaps

**Review by ritave — CHANGES_REQUESTED** [review]
Minor problems:
1. The ones in the comments
2. The newlines at the end of the files are inconsistent. Please add them everywhere.
3. Might as well add a link in a README.md to the Github Pages

Questions:
1.  Is it possible to hide Jekyll files to a sub-folder so that they don't pollute root?

File: _layouts/sip.html (hunk 3 above)
Comment (FrederikBolding) [reply _layouts/sip.html:76]: Just included from the EIP/CAIP template. I can remove it if you prefer

File: _layouts/sip.html (hunk 2 above)
Comment (ritave) [reply _layouts/sip.html:41]: Sure, sounds useful

File: _layouts/sip.html (hunk 3 above)
Comment (ritave) [reply _layouts/sip.html:63]: No opinion, up to you.

**Review by ritave — APPROVED** [review]

**Pull request merged**